src/power_grid_model/_core/utils.py

```diff
@@ -651,10 +651,8 @@ def _extract_columnar_data(
     """
     not_columnar_data_message = "Expected columnar data"
 
-    if is_batch is not None:
-        allowed_dims = [2, 3] if is_batch else [1, 2]
-    else:
-        allowed_dims = [1, 2, 3]
+    if_is_batch = [2, 3] if is_batch else [1, 2]
+    allowed_dims = (if_is_batch) if is_batch is not None else [1, 2, 3]
```

Comment:
although the SIM suggestion is better than the original, this still feels like it can be rewritten in a waaayyyy better way.

E.g. try
```suggestion
    if is_batch is None:
        allowed_dims = [1, 2, 3]
    elif is_batch:
        allowed_dims = [2, 3]
    else:
        allowed_dims = [1, 2]
```
now please run the `SIM` check again after that.

This is a good example of why such a linter is important (we actually get suggestions) and then our next step is t always double-check whether this really is the right path going forward